### PR TITLE
Remove PyArrow deprecation warning

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/__init__.py [hunk 1]
```diff
@@ -190,36 +191,17 @@
...
-    if pa_not_found:
-        pa_msg = "was not found to be installed on your system."
-    else:
-        pa_msg = (
-            f"was too old on your system - pyarrow {VERSIONS['pyarrow']} "
-            "is the current minimum supported version as of this release."
-        )
-
+# GH#55043 - deprecation of the data_manager option
+if "PANDAS_DATA_MANAGER" in os.environ:
```
Comment (mroeschke) [inline pandas/__init__.py:195]: If targeting main, this should probably remain removed

Comment (phofl) [reply pandas/__init__.py]: thx. done

File: doc/source/whatsnew/v2.2.1.rst [hunk 2]
```diff
@@ -60,6 +60,16 @@ Bug fixes
...
 Other
 ~~~~~
+
+.. note::
+
+    The ``DeprecationWarning`` that was raised when pandas was imported without PyArrow being
+    installed has been removed. This decision was made because the warning was too noisy for too
+    many users and a lot of feedback was collected about the decision to make PyArrow a required
+    dependency. Pandas is currently considering the decision whether or not PyArrow should be added
+    ad a hard dependency in 3.0. Interested users can follow the discussion
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.1.rst:70]:
```suggestion
    as a hard dependency in 3.0. Interested users can follow the discussion
```

**Review by mroeschke — APPROVED** [review]

**Comment by lithomas1** [comment]
thanks @phofl.

**Comment by phofl** [comment]
I'll do the backport

**Comment by phofl** [comment]
#57558

**Pull request merged**